Sources/SafeDICore/Models/TypeDescription.swift

```diff
@@ -425,6 +429,38 @@ extension ExprSyntax {
     }
 }
 
+// MARK: - UnorderedComparingArray
+
+public struct UnorderedComparingArray<Element: Codable & Hashable & Sendable>: Codable, Hashable, Sendable, Collection {
```

Comment:
What do you think about changing the name? Having a subscript for an unordered array seems a bit confusing.

Reply:
All sequences have subscripts, and sequences are not necessarily ordered logically (though their order is stable). It's a quirk of Foundation.

Reply:
But I am super open to other names

Reply:
Going with `UnorderedComparingCollection`